languages/python/concepts/dicts/about.md

```diff
@@ -0,0 +1,34 @@
+A dictionary is a data structure that associates a key to a value. In Python, it's called the [mapping type dict][mapping-types-dict], this data type enables to retrieve the value object quickly knowing the key. Comparing with searching an object in the items a list, a dictionary would use more memory, but would be faster, especially for large collections.
+
+## Keys and Values
+
+A dictionary can be though of as a simple key:value pair mapping, it can be declared using `{k1: v1, k2: v2}` or `dict(k1=v1, k2=v2)`, but there is much more ways of creating and initializing a dictionary including dict comprehensions or other constructor parameters that are illustrated on the [official docs][mapping-types-dict]. Inserting a key:value pair can be done with `dict[key] = value` and the retrieval of a value by `v = dict[k]`.
+
+A key is unique across a dictionary, but the value associated with it can be replaced, updated or altered as long as that key exists. To be a valid key, the key should be [hashable][term-hashable], hashable keys include for example numbers, strings or tuples of immutable values.
+
+A value can be of any data type, including Build-in types, custom types or complex objects. Common used objects are: numbers, string, list, dictionary or sets.
+
+## Methods
+
+Dictionaries implement various methods to allow easy initializing, updates and viewing, for example getting a value from a specific key in the dictionary, setting a default value for a key and more. Dictionaries are also iterable by their keys, values or key-value pairs.
+
+Some useful `dict` methods that are useful:
+
+- Get a value from a dictionary by using the `.get(key, [default])` returns the value for a given key, would not error and returns the `default` value if the key is missing.
+- Set a default value for a key with the `.setdefault(key, [default])` method, which is similar to the `.get()`, but it additionally inserts that key with the default value if it does not exist inside the dictionary.
```

Comment:
```suggestion
- Set a default _value_ for a given key with the `.setdefault(key, [default])` method.  This can be viewed as a corollary to the `.get()` method -- `.setdefault(key, [default])` will set an _existing_ key to the default value given, but will additionally _create that key_ with the default value if it does not already exist inside the dictionary.  
```